pipeline.groovy

```diff
@@ -97,19 +97,35 @@ def mapSteps (names, action) {
 *******************************************************************************/
 
 def getSources (name) {
+    def git_ref = 'master'
+    def pr_repo
+    def pr_id
+
     // presence of CHANGE_URL environment variable means this pipeline tests
     // Pull Request and has to checkout PR branch instead of master branch
     // for relevant repository:
     def regex = /https:\/\/github.com\/[^\/]+\/([^\/]+)\/pull\/(\d+)/
     def match = (env.CHANGE_URL =~ regex)
-    def pr_repo = match ? match[0][1] : ""
+    if (match) {
+        pr_repo = match[0][1]
+        pr_id = match[0][2]
+    }
     match = null
 
+    if (pr_repo && pr_id) {
+        def api_url = "https://api.github.com/repos/dlang/$pr_repo/pulls/$pr_id"
+        def extract_cmd = /python3 -c "import sys, json; print(json.load(sys.stdin)['base']['ref'])"/
```

Comment:
Can use [jq](https://stedolan.github.io/jq/) for such stuff, already preinstalled.
`jq -r '.base.ref'`
Also see https://github.com/dlang/dmd/blob/4f1425c1c7b07ffd72997151c614d51f63ddc082/circleci.sh#L55.